### DISCUSSION TITLE
Session not been set - Update 10 to 11

### DISCUSSION THREAD

**Discussion by sngm** [pimcore/pimcore#16602] [Q&A / Support] [answered]
Hello folks!

After updating from version 10 to 11, I now get the following error message:

"Uncaught PHP Exception Symfony\Component\HttpKernel\Exception\BadRequestHttpException: "Session has not been set." at HttpKernel.php line 83"

![image](https://github.com/pimcore/pimcore/assets/1529090/14d66629-9c3a-48d9-8d47-543ae7821505)

I am at a loss as to what I could check to fix the problem. Do you have an idea or a tip for me on how to proceed? Can I share any more information that will help you fix the problem?

I have followed the update instructions in the documentation as far as I have understood correctly. I have also tried to empty the cahce several times. I am pretty much at a loss.

Thank you for any help.

Greetings

Magnus

**Comment by sngm** [comment]
I have now copied the index.php from the demo repository into my project. As mine deviated from this.

Since an autoload_runtime was referenced there, I have now installed symfony/runtime for it. This then created an autoload_runtime.php. I only had one autoload.php so far.

But I still get almost the same error message:

![image](https://github.com/pimcore/pimcore/assets/1529090/2a244184-bda3-4533-a1a8-6d04ac530b85)

So I still need help on what I can check to get the project working again. Thank you already.

Magnus

**Comment by BigJabber** [comment]
I've run into the same issue as you this week. Did you manage to get past this at all?

**Reply by sngm** [reply]
Unfortunately, I have not yet been able to solve it. Were you able to find a solution?

**Reply by BigJabber** [reply]
No - I backed out and reverted back to my 10.6 install. 

I'm going to retry the install in the next couple of weeks, I'll drop an update if I work out what the issue was.

**Reply by dpfaffenbauer** [reply]
did you update the security config too?

**Reply by BigJabber** [reply]
I did. As far as I could tell I'd changed everything I needed to but presumably I missed something. Is there a specific part of the security config you think that would relate to?

**Reply by dpfaffenbauer** [reply]
I'd just copy the new security.yaml from pimcore 11. if you did that, I don't know what you might missed.

**Reply by sngm** [reply]
@dpfaffenbauer First of all, thanks for your interest in this problem. lol. Is the security.yaml the only file that causes this problem? I'm just wondering because you ask directly for this file. Can there be no other causes?
I copied this file from the Pimcore git into my project (from the demo repository).  The result remains the same. I am more than perplexed at the moment. I have already restarted the update 3-4 times and always end up with this error. Do you have any tips on what I could debug or test to narrow down the problem further?

**Reply by dpfaffenbauer** [reply]
It can be caused by many things. Your symfony session config could also be the issue.

**Comment by benwalch** [comment, marked as answer]
after an upgrade from 10 to 11, this is mainly caused by pimcore's `MaintenancePageListener`.

You have to remove the `kernel.event_listener` tag, since pimcore switched it on 11.x to be an event subscriber.
Having the tag with the wrong priority causes this listener to be called too early, when no session has been initialized.

Pimcore's docs about this are (still) wrong (a shame): https://docs.pimcore.com/platform/Pimcore/Extending_Pimcore/Maintenance_Mode/

hope this helps.

**Reply by jdreesen** [reply]
Maybe you want to provide a PR which fixes the docs?
